libcontainer/integration/execin_test.go

```diff
@@ -30,16 +30,37 @@ func TestExecIn(t *testing.T) {
 	// Execute a first process in the container
 	stdinR, stdinW, err := os.Pipe()
 	ok(t, err)
+	defer func() {
+		_ = stdinR.Close()
+		_ = stdinW.Close()
+	}()
+	stdoutR, stdoutW, err := os.Pipe()
+	ok(t, err)
+	defer func() {
+		_ = stdoutR.Close()
+		_ = stdoutW.Close()
+	}()
+
+	ch := waitStdOut(stdoutR)
 	process := &libcontainer.Process{
-		Cwd:   "/",
-		Args:  []string{"cat"},
-		Env:   standardEnvironment,
-		Stdin: stdinR,
-		Init:  true,
+		Cwd:    "/",
+		Args:   []string{"cat", "/proc/self/cmdline", "-"},
+		Env:    standardEnvironment,
+		Stdin:  stdinR,
+		Stdout: stdoutW,
+		Init:   true,
 	}
 	err = container.Run(process)
-	_ = stdinR.Close()
-	defer stdinW.Close() //nolint: errcheck
+	defer func() {
+		_, _ = stdinW.Write([]byte("hello"))
```

Comment:
why do we write this? To stop the go routine? Can you add a comment or the message written can be self-explanatory instead of this hello?

If it's that, is it needed? Won't the close of the pipe already free the goroutine?